File: hw/pazmino/u2/HW27UnitTestPython/HW27Calculator/basic_operation_controller.py
from basic_operation import BasicOperation

class BasicOperationController:
    def __init__(self, basic_operation):
        self.basic_operation = basic_operation

    def add(self, addend1=None, addend2=None):
        if addend1 is not None and addend2 is not None:
            return addend1 + addend2
        else:
            return self.basic_operation.get_term1() + self.basic_operation.get_term2()

    def get_basic_operation(self):
        return self.basic_operation

    def set_basic_operation(self, basic_operation):
        self.basic_operation = basic_operation

File: ws/charij/u2/WS24_UNIT_TEST/Calculator/src/main/java/ec/edu/espe/calculator/controller/BasicOperationController.java


package ec.edu.espe.calculator.controller;

import ec.edu.espe.calculator.model.BasicOperation;

/**
 *
 * @author Sebastian Charij
 */
public class BasicOperationController {
    private BasicOperation basicOperation;
    
    public float add(float addend1, float addend2){
        float sum;
        sum = addend1 + addend2;
        return sum;
    }
    public float add(){
        float sum;
        sum = basicOperation.getTerm1() + basicOperation.getTerm2();
        return sum;
    }

    public BasicOperationController(BasicOperation basicOperation) {
        this.basicOperation = basicOperation;
    }
    

    /**
     * @return the basicOperation
     */
    public BasicOperation getBasicOperation() {
        return basicOperation;
    }

    /**
     * @param basicOperation the basicOperation to set
     */
    public void setBasicOperation(BasicOperation basicOperation) {
        this.basicOperation = basicOperation;
    }
    
    
}
